lib/node_modules/@stdlib/string/base/replace-before/benchmark/benchmark.js

```diff
@@ -0,0 +1,54 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2023 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var bench = require( '@stdlib/bench' );
+var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
+var fromCodePoint = require( '@stdlib/string/from-code-point' );
+var pkg = require( './../package.json' ).name;
+var replaceBefore = require( './../lib' );
+
+
+// MAIN //
+
+bench( pkg, function benchmark( b ) {
+	var out;
+	var str;
+    var replacement;
+	var i;
+
+	str = 'To be, or not to be, that is the question.';
+    replacement = 'foo';
+
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		out = replaceBefore( str, fromCodePoint( i%126 ), replacement );
+		if ( typeof out !== 'string' ) {
+			b.fail( 'should return a string' );
+		}
+	}
+	b.toc();
+	if ( !isString( out ) ) {
+		b.fail( 'should return a string' );
+	}
+	b.pass( 'benchmark finished' );
+	b.end();
+});
```

Comment:
All files should have a trailing newline. Make sure you've setup EditorConfig for your IDE, as the editor can handle this automatically based on our EditorConfig configuration.

Reply:
See the dev guide for more info.